CMakeLists.txt

```diff
@@ -9,7 +9,11 @@ project(cime_tests Fortran C)
 
 # We rely on pio for cmake utilities like findnetcdf.cmake, so that we don't
 # need to duplicate this cmake code
-list(APPEND CMAKE_MODULE_PATH "${CIME_ROOT}/src/externals/pio2/cmake")
+if (EXISTS ${SRC_ROOT}/libraries/parallelio/cmake)
+  list(APPEND CMAKE_MODULE_PATH "${SRC_ROOT}/libraries/parallelio/cmake")
+else()
+  list(APPEND CMAKE_MODULE_PATH "${CIME_ROOT}/src/externals/pio2/cmake")
+endif()
```

Comment:
Should we put the path in config_files.xml?  Same for MCT below.

Reply:
No code was removed.  I have no issues with adding the paths to config_files.xml.  I would want to update several other files to remain consistent.  So I would prefer to do that in a separate PR.  But if you want, I'll do it for this PR.

Reply:
After telecon:  Nevermind.  Each model can effectively set the location in config_files.xml already, this file is only for the cime unit tests and the logic will be removed after the old code is removed.